File: backend/PulseGym.Logic/DTO/WorkoutRequestDTO/WorkoutRequestViewDTO.cs
using PulseGym.Entities.Enums;

namespace PulseGym.Logic.DTO
{
    public class WorkoutRequestViewDTO
    {
        public Guid Id { get; set; }

        public DateTime DateTime { get; set; }

        public required WorkoutRequestStatus Status { get; set; }

        public required ClientViewDTO Client { get; set; }

        public required TrainerViewDTO Trainer { get; set; }
    }
}

File: frontend/src/app/models/workout-request/workout-request-view-dto.ts
import { WorkoutRequestStatus } from "src/app/enums/workout-request-status";
import { ClientViewDTO } from "../client/client-view-dto";
import { TrainerViewDTO } from "../trainer/trainer-view-dto";

export interface WorkoutRequestViewDTO {
    id: string;
    dateTime: string;
    status: WorkoutRequestStatus;
    trainer: TrainerViewDTO;
    client: ClientViewDTO;
}